crates/paralegal-policy/src/context.rs

```diff
@@ -322,7 +322,18 @@ impl Context {
 
     /// Returns whether the given Node has the marker applied to it directly or via its type.
     pub fn has_marker(&self, marker: Marker, node: Node) -> bool {
-        // TODO: check the type as well
+        if let Some(src) = node.typ.as_data_source() {
+            if let Some(typ) = &self.desc.controllers[&node.ctrl_id].types.get(&src) {
+                if typ.iter().any(|t| {
```

Comment:
This doesn't have to be an `if`, you can just return this expression. :)

Reply:
We don't want to return if it's false!

Reply:
Oh sorry. I did not realize the context. I thought the function ended in 336.